packages/wonder-blocks-modal/components/modal-launcher/modal-launcher-portal.test.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+import React from "react";
+import {mount} from "enzyme";
+
+import ModalLauncherPortal from "./modal-launcher-portal.js";
+
+describe("ModalPortal", () => {
+    test("Mounts its children in document.body", (done) => {
+        // Once the child mounts, check that it was mounted directly-ish into
+        // `document.body`, and finish the test.
+        const childrenRef = (element) => {
+            if (element) {
+                // Find the nearest parent, disregarding nodes created by
+                // ModalLauncherPortal and by `ReactDOM.render`.
+                let parent = element.parentNode;
+                while (
+                    parent.hasAttribute("data-modal-launcher-portal") ||
+                    parent.hasAttribute("data-reactroot")
```

Comment:
You could use `querySelector` to find this element and then use `.contains()` to check if the root contains `element`.

Reply:
If I'm understanding right, I don't think that's quite the same assertion: I want to confirm that this element is mounted _directly_ into the body, modulo some expected wrapper elements. If we just asserted that the element is a descendant of `document.body`, then this test would probably pass even without portaling.

Still, there might be a clearer way to phrase this assertion, or at least leave clearer comments? I'll give it a readability pass.

Reply:
Oh, I misunderstood your comment! Right, I'm also not trying to assert that there _is_ a wrapper element and that this is a descendant of it; I'm trying to assert that this is mounted as near to the body as possible.

Reply:
I've tried adding a few more comments to clarify the intent.

Reply:
But if there's only one element that has `data-reactroot` and `data-modal-launcher-portal` then my approach would work.  Since that's a pretty big if, safer to go with your approach.

Reply:
I'm not sure that's strictly true? Like, I think the following implementation would pass the alternate version of the test:

```
render() {
    return <div data-modal-launcher-portal>{this.props.children}</div>;
}
```

If I'm understanding correctly, we'd search for the `data-modal-launcher-portal` element, check whether the children are in it, and they are so it would pass. But this component doesn't actually portal the content to the root of the page; it just renders it inline.